[shishirbychapur] iP

shishirbychapur opened the pull request:
## Jarvis, Your Personal Task Manager 🤖!

> "Simplicity is the ultimate sophistication." - Leonardo da Vinci [(source)](https://www.theorderexpert.com/11-inspirational-quotes-about-organization/)

How do you get started?

- Download the latest  ` .jar ` file from [here.](https://github.com/shishirbychapur/ip/releases)
- Navigate to the directory of installation, and open Terminal.
- Run ` java -jar Duke.jar ` and enjoy the luxury of having your own manager!

What are some of the features?

1. Allows various types of tasks (Events, Deadlines, Todo)
2. Stores and retrieves data directly to/from a designated .txt file
3. Enables marking/unmarking of tasks on completion
4. Grants users the ability to search for their respective task

Why choose Jarvis?

- [x] **Easy** to learn
- [x] **Available** _24/7_ at your service
- [x] Created by ~~Shishir~~ **Tony Stark** himself
- [x] **Free** of charge

Here's a sneak peak of the java code used to develop Jarvis!

```java
    /**
     * Initialises Jarvis
     * @param args Input args.
     */
    public static void main(String[] args) {
        Duke bot = new Duke("data/tasks.txt");
        bot.interact();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

shishirbychapur committed:
Rename, Greet, Exit

shishirbychapur committed:
Add Echo

shishirbychapur committed:
Add/List

shishirbychapur committed:
Mark as Done

shishirbychapur committed:
ToDo, Event, Deadline

shishirbychapur committed:
Setup Automated Text UI Testing

shishirbychapur committed:
Handle Errors

shishirbychapur committed:
Delete

shishirbychapur committed:
Add javadocs

shishirbychapur committed:
Add File Compatibility

shishirbychapur committed:
Merge branch 'branch-Level-7'

shishirbychapur committed:
Add Date/Time Functionality

shishirbychapur committed:
Merge branch 'branch-Level-8'

shishirbychapur committed:
Fix Routing

shishirbychapur committed:
Improve OOP

shishirbychapur committed:
Merge branch 'branch-More-OOP'

shishirbychapur committed:
Divide Classes into Packages

shishirbychapur committed:
Merge branch 'branch-Packages'

shishirbychapur committed:
Merge branch 'add-gradle-support' of https://github.com/shishirbychapur/ip

shishirbychapur committed:
Add Gradle Support

shishirbychapur committed:
Add Unit Tests

shishirbychapur committed:
Package as JAR File

shishirbychapur committed:
Improve Javadocs

shishirbychapur committed:
Handle Find Command

shishirbychapur committed:
Merge branch 'branch-Level-9'

shishirbychapur committed:
Merge branch 'branch-A-JavaDoc'

shishirbychapur committed:
Improve JavaDocs

shishirbychapur committed:
Merge branch 'master' into branch-A-CodingStandard

shishirbychapur committed:
Fix Incorrect Field Bug

shishirbychapur committed:
Merge branch 'branch-A-CodingStandard'

shishirbychapur committed:
Improve Code Standard

shishirbychapur committed:
Fix Coding Standards + Improve Code

shishirbychapur committed:
Merge branch 'master' into branch-A-CodingStandard

shishirbychapur committed:
Fix Naming Issue

shishirbychapur committed:
Merge branch 'branch-A-CodingStandard'

shishirbychapur committed:
Setup Checkstyle + Fix Indentation Errors

shishirbychapur committed:
Merge branch 'branch-A-CheckStyle'

shishirbychapur committed:
Setup JavaFx

shishirbychapur committed:
Integrate JavaFx + Refactor Code

shishirbychapur committed:
Merge branch 'branch-Level-10'

shishirbychapur committed:
Fix Code Standard

shishirbychapur committed:
Improve GUI

shishirbychapur committed:
Change DP to Circle

shishirbychapur committed:
Change DP to Circle

supermii2 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,69 @@
+package duke;
```
I think references with Duke should be renamed with Jarvis

supermii2 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+        }
+        return c;
+    }
+
```
I think having all of the validation checks in one section is pretty confusing, especially when most of them are exclusive to one command! Perhaps use parser to handle the input into a form readable by each command, and then handle the validation there?

supermii2 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+            return new IncorrectCommand("Please enter a valid task.");
+        }
+
+        if (split[0].equals("deadline")) {
```
I think addition of new task types might make this section complicated an error prone. (Imagine if you have a typo in the word "deadline", and it only appears here and not in the switch-case statement). Ditto this with the remaining sections.

supermii2 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,165 @@
+        if (split.length != 1) {
+            return new IncorrectCommand("Please enter a valid command!");
+        }
+        return flag ? new ExitCommand() : new ListCommand();
```
I'm not a fan of using ```flag``` to choose between two very different commands. I think it would be better to make this more intuitive.

supermii2 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,69 @@
+            return exc.getMessage();
+        }
+    }
+
```
I think this whitespace/line might not be necessary!

supermii2 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,102 @@
+
+    /**
+     * Constructs the AddCommand Object.
+     * @param description Description of the task.
```
I think you're missing the @param for the deadline time in the javadoc

supermii2 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,102 @@
+    @Override
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task task;
+        switch (this.type) {
```
Good job on the formatting of switch-case statements! This tripped me up many times

supermii2 commented on src/main/java/duke/commands/AddCommand.java:
```diff
@@ -0,0 +1,102 @@
+    public String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        Task task;
+        switch (this.type) {
+        case "todo":
```
Might be worth using an enum to handle the different types! In the event you have to add more types of tasks down the line, it will save headaches.

supermii2 reviewed (commented):
Overall the code follows coding standards and is pretty readable! I appreciate the use of new lines to separate different functions to ease readibility, which I am going to use for my code too.

shishirbychapur committed:
Change Class Name to Jarvis

shishirbychapur committed:
Set up CI/CD

shishirbychapur committed:
Fix Checkstyle Violations

shishirbychapur committed:
Fix Coding Violations in test folder

shishirbychapur committed:
Fix Test

shishirbychapur committed:
Change YML File

shishirbychapur committed:
Add Assertions

Assertions have been added to ensure important assumptions hold true.
For example, task class should not be recieving empty description
strings and exception messages should not be blank.

shishirbychapur committed:
Fix Testcases

shishirbychapur committed:
Fix Checkstyle Error

shishirbychapur committed:
Merge pull request #1 from shishirbychapur/branch-A-Assertions

Add Assertions

shishirbychapur committed:
Fix Code To Follow Coding Standards

The parser classes had functions with >30 LoC, and functions not
following the KISS principle. Also made the happy path prominent, and
added more abstractions, and removed deep nesting.

shishirbychapur committed:
Merge pull request #2 from shishirbychapur/branch-A-CodeQuality

Fix Code To Follow Coding Standards

shishirbychapur committed:
Add Reminder Feature + Improve UI

Added a reminder class to incorporate the reminder command. Improved the
UI of the messages by adding indentations and emojis.

shishirbychapur committed:
Merge pull request #3 from shishirbychapur/branch-BCD-Extension

Add Reminder Feature + Improve UI

shishirbychapur committed:
Improve GUI

shishirbychapur committed:
Merge branch 'branch-A-BetterGui'

shishirbychapur committed:
Fix Unit Tests + Code Quality Issues

shishirbychapur committed:
Add UiMessages + ExceptionMessages Class

shishirbychapur committed:
Add Enumerations

shishirbychapur committed:
Improve Code Quality

shishirbychapur committed:
Add Representative Screenshot

shishirbychapur committed:
Setup up User Guide

shishirbychapur committed:
Fix Small Issue with UG

shishirbychapur committed:
Update UG

Add description of Jarvis.

shishirbychapur committed:
Update README.md

shishirbychapur committed:
Create _config.yml

shishirbychapur committed:
Improve UG

shishirbychapur committed:
Update README.md

shishirbychapur committed:
Update _config.yml

shishirbychapur committed:
Update README.md

shishirbychapur committed:
Add Finishing Touches for UG

shishirbychapur committed:
Add Final Release

shishirbychapur committed:
Fix Bug

shishirbychapur committed:
Add Compatibility for Linux
